core/server/services/auth/session/middleware.js

```diff
@@ -0,0 +1,123 @@
+const {URL} = require('url');
+const common = require('../../../lib/common');
+const config = require('../../../config');
+const settingsCache = require('../../settings/cache');
+const models = require('../../../models');
+const session = require('express-session');
+const SessionStore = require('./store');
+const urlService = require('../../url');
+
+const getOrigin = (req) => {
+    const origin = req.get('origin');
+    const referrer = req.get('referrer');
+
+    if (!origin && !referrer) {
+        return null;
+    }
+
+    if (origin) {
+        return origin;
+    }
+
+    try {
+        return new URL(req.get('referrer')).origin;
+    } catch (e) {
+        return null;
+    }
+};
+
+let UNO_SESSIONIONA;
+const getSession = (req, res, next) => {
+    if (!UNO_SESSIONIONA) {
+        UNO_SESSIONIONA = session({
+            store: new SessionStore(models.Session),
+            secret: settingsCache.get('session_secret'),
+            resave: false,
+            saveUninitialized: false,
+            name: 'ghost-admin-api-session',
+            cookie: {
+                maxAge: 184 * 24 * 60 * 60 * 1000, // number of days in second half of year
```

Comment:
We have a config for that in lib/constants

Reply:
Ah wicked :+1: